### PR TITLE
chore: cleanup rust code (except for modules)

### PR DESCRIPTION
**Pull request by tvolk131**
I temporarily enabled clippy pedantic rules and fixed the issues it caught (except the ones that sacrifice code readability)

### CODE REVIEW

**Comment by elsirion** [comment]
CI failure was unrelated, but needs rebase now.

**Comment by tvolk131** [comment]
Markdown check failed because of a nix bug (see #5379) but everything else looks good!

File: fedimint-server/src/net/queue.rs [hunk 1]
```diff
@@ -51,12 +51,7 @@ where
 
     pub fn ack(&mut self, msg_id: MessageId) {
         debug!("Received ACK for {:?}", msg_id);
-        while self
-            .queue
-            .front()
-            .map(|msg| msg.id <= msg_id)
-            .unwrap_or(false)
-        {
+        while self.queue.front().is_some_and(|msg| msg.id <= msg_id) {
```
Comment (elsirion) [inline fedimint-server/src/net/queue.rs:54]: TIL

**Comment by elsirion** [comment]
@dpc @joschisan How much value do you see in these kinds of cleanups? On the one hand, there are some really neat changes in there, otoh it's a lot of churn and not enforced in CI, so will deteriorate again over time.

**Comment by elsirion** [comment]
Maybe we can enforce this in CI and just `#[ignore(clippy::…)]`?

**Comment by dpc** [comment]
> Maybe we can enforce this in CI and just `#[allow(clippy::…)]`?

If there are some clippy lints that we find valuable that are not enabled be default, we should enable them, sure @tvolk131 @elsirion :+1:

**Comment by tvolk131** [comment]
> Maybe we can enforce this in CI and just `#[allow(clippy::…)]`?

Done! See #5389

**Pull request merged**